docs/guide/counter/plugins/counter/counter.go

```diff
@@ -1,98 +1,197 @@
 package counter
 
 import (
-	"fmt"
+	rawerr "errors"
 
 	abci "github.com/tendermint/abci/types"
-	"github.com/tendermint/basecoin/types"
 	"github.com/tendermint/go-wire"
+
+	"github.com/tendermint/basecoin"
+	"github.com/tendermint/basecoin/errors"
+	"github.com/tendermint/basecoin/modules/coin"
+	"github.com/tendermint/basecoin/stack"
+	"github.com/tendermint/basecoin/types"
 )
 
-type CounterPluginState struct {
-	Counter   int
-	TotalFees types.Coins
+// CounterTx
+//--------------------------------------------------------------------------------
+
+// register the tx type with it's validation logic
+// make sure to use the name of the handler as the prefix in the tx type,
+// so it gets routed properly
+const (
+	NameCounter = "cntr"
+	ByteTx      = 0x21
```

Comment:
Okay in my ideal world we actually never use a byte only a type byte variable which would look something like this:
``` golang
const (
    TBTx = iota
    TBHandler
    TBRigelIsCool
)
```
These will automatically assign 0x00, 0x01, 0x02, there are ways to make it start the count at any byte. This way we are never confused about 0x21 is supposed to mean, because we have a description of what that byte is indicating via the variable name